feat(e2e): Added E2E testing

rolaveric opened the pull request:
Tests are defined in `*.scenario.*` files.
Data fixtures defined with `ngMockE2E:$httpBackend.when()` calls are defined in `*.fixture.*` files.
Data fixtures are only included in `build`, for development, not `compile`.
The Karma ng-scenario adapter is statically included as a temporary fix until it can be included as a sibling dependency:
https://github.com/karma-runner/grunt-karma/issues/13

rolaveric committed:
feat(e2e): Added E2E testing
Tests are defined in `*.scenario.*` files.
Data fixtures defined with `ngMockE2E:$httpBackend.when()` calls are defined in `*.fixture.*` files.
Data fixtures are only included in `build`, for development, not `compile`.
The Karma ng-scenario adapter is statically included as a temporary fix until it can be included as a sibling dependency:
https://github.com/karma-runner/grunt-karma/issues/13

joshdmiller commented:
@rolaveric - I like this. Thanks! In going to take a more detailed look this week, but I hope to merge this soon.

rolaveric committed:
fix(*): E2E karma server clashing with unit server
It appears the port number for the karma server running unit tests in the background during 'grunt watch' was the same as the one used when running E2E tests.  As a result, the 'karma:unit:run' test wasn't running properly.

revolunet commented:
awesome :)

rolaveric commented:
Closed PR as (per comments from #39 and #196 ) ngbp will be using protractor for E2E testing, making this change redundant.
